$NetBSD: patch-ab,v 1.5 2008/11/09 20:09:03 mwdavies Exp $

libamarok unconditionally uses libdl.
http://bugs.kde.org/show_bug.cgi?id=168741

Use newer pkg-config support from cmake.

--- CMakeLists.txt.orig	2008-10-31 14:23:36.000000000 +0000
+++ CMakeLists.txt
@@ -26,6 +26,8 @@ add_definitions(${QT_DEFINITIONS} ${KDE4
 include(CheckLibraryExists)
 check_library_exists(dl dlopen "" LIBDL_FOUND)
 
+check_library_exists(dl dlopen "" LIBDL_FOUND)
+
 set(TAGLIB_MIN_VERSION "1.5")
 find_package(Taglib)
 macro_log_feature( TAGLIB_FOUND "taglib" "Read and write tags in audio files" "http://www.freshmeat.net/projects/taglib" TRUE "1.5.0" "")
@@ -52,10 +54,12 @@ include(CheckTagLibFileName)
 macro_optional_find_package(GLIB2)
 macro_log_feature( GLIB2_FOUND "glib2" "Required by libgpod and mp3tunes" "http://www.gtk.org" FALSE "2.x" "")
 
-macro_optional_find_package(Gdk)
+INCLUDE(FindPkgConfig)
+pkg_check_modules(GDK gdk-pixbuf-2.0)
 macro_log_feature( GDK_FOUND "Gdk" "Support for artwork on iPod audio devices via GdkPixbuf" "http://developer.gnome.org/arch/imaging/gdkpixbuf.html" FALSE "2.0.x" "" )
 
-macro_optional_find_package(Ipod)
+#macro_optional_find_package(Ipod)
+pkg_check_modules(IPOD libgpod-1.0)
 macro_log_feature( IPOD_FOUND "libgpod" "Support Apple iPod audio devices" "http://sourceforge.net/projects/gtkpod/" FALSE "0.6.0" "" )
 
 #macro_optional_find_package(MySQL)
@@ -86,7 +90,8 @@ macro_log_feature( _mp3tunes_crypto "ope
 find_package(GObject)
 macro_log_feature( GOBJECT_FOUND "gobject" "Required by mp3tunes." "http://www.gtk.org" FALSE "2.x" "" )
 
-macro_optional_find_package(Loudmouth)
+#macro_optional_find_package(Loudmouth)
+pkg_check_modules(LOUDMOUTH loudmouth-1.0)
 macro_log_feature( LOUDMOUTH_FOUND "loudmouth" "Loudmouth is the communication backend needed by mp3tunes for synching." "http://www.loudmouth-project.org" FALSE "" "" )
 
 find_package(KdeMultimedia)
